src/etc/inc/interfaces.inc

```diff
@@ -2357,6 +2358,87 @@ function interface_virtual_create($interface)
     }
 }
 
+function interface_dslite_configure($verbose, $parent = 'wan', $aftr)
+{
+    global $config;
+
+    // Should be tested outside, but let's make sure we won't mess around
+    if (!interface_dslite_is_enabled($parent)) {
+        return;
+    }
+
+    $wanconfig = $config['interfaces'][$parent];
+    $mtu = !empty($wanconfig['dslite_tunnel_mtu']) ? $wanconfig['dslite_tunnel_mtu'] : '';
+    $ip = get_interface_ipv6($parent);
+
+    if ($verbose) {
+        log_error(sprintf("Starting DS-Lite tunnel with AFTR '%s' on interface %s with external IP '%s'", $aftr, $parent, $ip));
+    }
+
+    $gifif = legacy_interface_create('gif', "{$parent}_dslite");
+
+    $pmtu = '';
+    if (!empty($mtu)) {
+        $pmtu = sprintf('mtu %s', $mtu);
+    }
+
+    mwexecf('/sbin/ifconfig %s inet6 tunnel %s %s %s -accept_rtadv ifdisabled', array($gifif, $ip, $aftr, $pmtu));
+    mwexecf("/sbin/ifconfig %s inet 192.0.0.2 192.0.0.1 netmask 255.255.255.248", array($gifif));
+
+    $dslite_tunnel_file = sprintf("/tmp/dslite_%s.state", $parent);
+    @file_put_contents($dslite_tunnel_file, $gifif);
+
+    $dslite_aftr_file = sprintf("/tmp/dslite_%s_aftr.state", $parent);
+    @file_put_contents($dslite_aftr_file, $aftr);
+}
+
+function interface_dslite_destroy($verbose, $parent = 'wan')
+{
+    global $config;
+
+    // Should be tested outside, but let's make sure we won't mess around
+    if (!interface_dslite_is_enabled($parent)) {
+        return;
+    }
+
+    $gifif = interface_dslite_find_tunnel_interface($parent);
+    if (!empty($gifif)) {
```

Comment:
could probably check for existence of the interface using `!empty(legacy_interfaces_details("{$parent}_dslite"))`

Reply:
After ditching find_tunnel, doesn't it make more sense to do a `    if (does_interface_exist($gifif)) {` ?

Reply:
you're probably right, both do an ifconfig call.